crates/aptos-crypto/src/p256/p256_sigs.rs

```diff
@@ -0,0 +1,191 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This file implements traits for ECDSA signatures over NIST-P256.
+
+use crate::{
+    p256::{P256PrivateKey, P256PublicKey, ORDER_HALF, ORDER_MINUS_ONE},
+    hash::CryptoHash,
+    traits::*,
+};
+use anyhow::{anyhow, Result};
+use aptos_crypto_derive::{DeserializeKey, SerializeKey};
+use core::convert::TryFrom;
+use serde::Serialize;
+use std::{cmp::Ordering, fmt};
+use signature::Verifier;
+use p256::NonZeroScalar;
+use p256::NistP256;
+use p256::elliptic_curve::Curve;
+
+use super::P256_SIGNATURE_LENGTH;
+
+/// A P256 signature
+#[derive(DeserializeKey, Clone, SerializeKey)]
+pub struct P256Signature(pub(crate) p256::ecdsa::Signature);
+
+impl private::Sealed for P256Signature {}
+
+impl P256Signature {
+    /// The length of the P256Signature
+    pub const LENGTH: usize = P256_SIGNATURE_LENGTH;
+
+    /// Serialize an P256Signature.
+    pub fn to_bytes(&self) -> [u8; P256_SIGNATURE_LENGTH] {
+        // The RustCrypto P256 `to_bytes` call here should never return a byte array of the wrong length
```

Comment:
Should or never does? Under what conditions does it return a byte array of the wrong length?

Reply:
See related reply in https://github.com/aptos-labs/aptos-core/pull/9594#discussion_r1341601992